debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/parser/LogMinerColumnResolverDmlParser.java

```diff
@@ -27,9 +28,38 @@ public LogMinerColumnResolverDmlParser(OracleConnectorConfig connectorConfig) {
     @Override
     protected int getColumnIndexByName(String columnName, Table table) {
         if (columnName.matches("COL \\d*")) {
+            final int columnIndex = Integer.parseInt(columnName.substring(4));
             // LogMiner always uses 1-based indices, 0-based indices are required.
-            return Integer.parseInt(columnName.substring(4)) - 1;
+            return (columnIndex + getGeneratedColumnCountSlicedByColumnIndex(table, columnIndex)) - 1;
```

Comment:
@Naros Is this something that can be cached per-table so it is not recomputed for every single record?

Reply:
Reworked code and added the cache.